client/state/reader/posts/test/actions.js

```diff
@@ -52,8 +63,103 @@ describe( 'actions', () => {
 					railcar: 'foo'
 				}
 			];
-			receivePosts( posts )( dispatchSpy );
+			actions.receivePosts( posts )( dispatchSpy );
 			expect( trackingSpy ).to.have.been.calledWith( 'calypso_traintracks_render', 'foo' );
 		} );
+
```

Comment:
I think we could benefit from another test here aimed at the `postToKey` function. Maybe something called "should read site post if external and has site id". Just something that tests that this logic is always respected: `post.is_external && post.site_ID ? post.site_ID : undefined`